tcf_website/views/browse.py

```diff
@@ -157,74 +154,57 @@ def course_instructor(request, course_id, instructor_id):
         (instructor.full_name, None, True)
     ]
 
-    data = {
-        # rating stats
-        "average_rating": safe_round(instructor.average_rating_for_course(course)),
-        "average_instructor": safe_round(instructor.average_instructor_rating_for_course(course)),
-        "average_fun": safe_round(instructor.average_enjoyability_for_course(course)),
-        "average_recommendability":
-            safe_round(instructor.average_recommendability_for_course(course)),
-        "average_difficulty": safe_round(instructor.average_difficulty_for_course(course)),
-        # workload stats
-        "average_hours_per_week": safe_round(instructor.average_hours_for_course(course)),
-        "average_amount_reading": safe_round(instructor.average_reading_hours_for_course(course)),
-        "average_amount_writing": safe_round(instructor.average_writing_hours_for_course(course)),
-        "average_amount_group": safe_round(instructor.average_group_hours_for_course(course)),
-        "average_amount_homework": safe_round(instructor.average_other_hours_for_course(course)),
-    }
+    data = Review.objects\
+        .filter(course=course_id, instructor=instructor_id)\
+        .aggregate(
+            # rating stats
+            average_rating=(
+                Avg('instructor_rating') +
+                Avg('enjoyability') +
+                Avg('recommendability')
+            ) / 3,
+            average_instructor=Avg('instructor_rating'),
+            average_fun=Avg('enjoyability'),
+            average_recommendability=Avg('recommendability'),
+            average_difficulty=Avg('difficulty'),
+            # workload stats
+            average_hours_per_week=Avg('hours_per_week'),
+            average_amount_reading=Avg('amount_reading'),
+            average_amount_writing=Avg('amount_writing'),
+            average_amount_group=Avg('amount_group'),
+            average_amount_homework=Avg('amount_homework'),
+        )
+    data = {key: safe_round(value) for key, value in data.items()}
 
     try:
         grades_data = CourseInstructorGrade.objects.get(
             instructor=instructor, course=course)
-
+    except ObjectDoesNotExist:  # if no data found
+        pass
+    # NOTE: Don't catch MultipleObjectsReturned because we want to be notified
```

Comment:
Is the idea that the site should just crash if this is the case? Is there a better way of logging this?

Reply:
I believe the standard approach is to [send emails to devs when there are server errors](https://docs.djangoproject.com/en/3.1/howto/error-reporting/#server-errors). Sendgrid is a popular service in the field, and it seems to allow [100 emails/day for free forever](https://sendgrid.com/pricing/). The free plan isn't so scalable but is probably sufficient for our use case, and this can be done in a separate PR.